src/HexagonOffload.cpp

```diff
@@ -909,9 +956,31 @@ Buffer<uint8_t> compile_module_to_hexagon_shared_object(const Module &device_cod
     auto bss = obj->find_section(".bss");
     if (bss != obj->sections_end()) {
         bss->set_alignment(128);
+        // TODO: We should set the type to SHT_NOBITS
+        // This will cause a difference in MemSize and FileSize like so:
+        //        FileSize = (MemSize - size_of_bss)
+        // When the Hexagon loader is used on 8998 and later targets,
+        // the difference is filled with zeroes thereby initializing the .bss
+        // section.
         bss->set_type(Elf::Section::SHT_PROGBITS);
+        debug(4) << "bss size = " << bss->get_size() << "\n";
+        Elf::Section::contents_iterator s = bss->contents_begin();
```

Comment:
Much less verbose to just do:

    std::fill(bss->contents_begin(), bss->contents_end(), 0);